detectable/src/main/java/com/synopsys/integration/detectable/detectables/npm/lockfile/parse/NpmLockfileGraphTransformer.java

```diff
@@ -86,30 +86,29 @@ private void transformTreeToGraph(NpmDependency npmDependency, NpmProject npmPro
         if (!shouldIncludeDependency(npmDependency)) {
             return;
         }
-        
+
         // add workspaces as direct dependencies
         if (workspaces != null && !StringUtils.isBlank(npmDependency.getName()) &&
-                workspaces.stream().anyMatch(x -> x.contains(npmDependency.getName()))) {
+                workspaces.stream().anyMatch(x -> x.equals(npmDependency.getName()))) {
             dependencyGraph.addDirectDependency(npmDependency);
             
             // add workspace requires
             for (NpmRequires required : npmDependency.getRequires()) {
                 Dependency workspaceDependency = lookupDependency(required.getName(), npmDependency, npmProject, externalDependencies);
                 dependencyGraph.addChildrenToRoot(workspaceDependency);
             }
-            
+        } else {
+            npmDependency.getRequires().forEach(required -> {
+                logger.trace(String.format("Required package: %s of version: %s", required.getName(), required.getFuzzyVersion()));
+                Dependency resolved = lookupDependency(required.getName(), npmDependency, npmProject, externalDependencies);
+                if (resolved != null) {
+                    logger.trace(String.format("Found package: %s with version: %s", resolved.getName(), resolved.getVersion()));
+                    dependencyGraph.addChildWithParent(resolved, npmDependency);
+                } else {
+                    logger.debug("No resolved dependency found for required package: {}", required.getName());
+                }
+            });
```

Comment:
Is this change relevant to the 2 points you mentioned in the PR description? Looks like simple refactoring to me to show that this code addresses the "no workspaces" situation - or is moving this under `else` significant in any other way?

Reply:
This code was here before and was executing without ill effects. I don't think it's really bad for example to put something in the dependencyGraph twice...I believe it just overwrites the previous placement. At any rate, I get the same behavior with and without the else but I just thought it was a small optimization not to execute this code again for non-workspace dependencies. But yeah it's not really critical so I didn't mention it in the points in the description of the PR.

Reply:
Cool. I was making sure if this was just an optimization or something else I didn't understand. Thank you 👍